### DISCUSSION TITLE
[Factions] Mechanic Machinist talent feedback from a completed MP campaign + resource balancing

### DISCUSSION THREAD

**Discussion by Udrakan** [FakeFishGames/Barotrauma#10588] [Unstable feedback]
This is my talent feedback for the entire 50+h campaign session that Wheatley hosted.
First, some raw info

- Hellish campaign (no meme)
- Vanilla
- No custom subs used
- server mostly at high population, sometimes even full
- mostly people from discord, we were all in the discord voip with mics muted
- no jovrad, but we kept going forward at a relatively steady pace without lingering

I'm going to be posting this from a POV of a mechanic Machinist, mostly doing quartermaster stuff, keeping an order in our items and resources except medical stuff & infantry weapons.

1. resource balancing
2. talents and class progression

### 1. resource balancing
As a quartermaster I could feel most of the resource balancing throughout the entire campaign. All of the mishaps, people crafting a fun thing they wanted despite supplies being low, impact of plants and going forward in the campaign, how much the wrecks give and so on. On top of that, I remember when Barotrauma had abundance of materials and ore in the levels and did several rebalances of it through modding with various successes and failures.

My overall conclusion is that the resource balance and recipe costs are good.

There is a severe lack of lead though. Lead was the sole resource we lacked almost the entire campaign and we even had to switch some of our sub guns to accomodate for the lack of lead in the levels and loot. Some infantry weapons had to be forbidden from use due to it as well, which was a detriment to fun. My suggestion is to have all outpost types sell it and have at least 10 [minavailable] of it. It's not a fun item to lack and try to get.

Next problem is headsets giving 2 plastic. Despite wiring and other crafting we have ended up with **over 150 plastic** while lacking other supplies. Copper was in extreme abundance too (over 100) despite several people including me wiring their own things, some with a fair amount of components and wires. Tin was the bottleneck (for FPGA and components).

### 2. talents and class progression
Me and banananaman (an engineer) have been playing the same initial sessions and both of us have reached the end talent in our main tree at the start of the aphotic plateau (we haven't been lingering, but kept going forward at a relatively steady pace).

My progression:
![image](https://user-images.githubusercontent.com/59519434/204116057-3a034dfa-3929-42f0-ba67-908f9e300bb4.png)
Heres my feedback for talents in the same order I unlocked them.

**Engine engineer**
![image](https://user-images.githubusercontent.com/59519434/204114854-05fcc6c7-8ec5-41d0-9294-c8c9e4c03119.png)
My first talent. Feels pretty important to finish missions faster. This talent suffers from a thing many other talents also suffer from. I'll call it "noticability". The talents, despite being passive, should somehow show clearly when they are applied - for example a small icon with this talent on the engine interface with % speed increase. Mechanics could also at a glance see if somebody already has this talent, which is important to know, since it doesn't stack.
suggestion: Icon of the talent appearing on the repair interface when talent is in effect, ideally showing % bonus applied (maybe on mouseover idk)

**Safety first**
![image](https://user-images.githubusercontent.com/59519434/204114871-4529efad-50e3-4949-92fb-9c2720214f67.png)
I love safety harness, but I'd like talents like these that just unlock a recipe to be more impactful. 
suggestion: Maybe Safety First and Modular Repairs could be coupled into 1 talent and another talent could be made anew. 
suggestion: Maybe have talent unlocked items sold for 20% more as a nice passive that goes with them.

**Oiled machinery**
![image](https://user-images.githubusercontent.com/59519434/204114930-d7357e1d-2656-4a01-9c48-817010759f18.png)
Feels mandatory for a mechanic whos managing supplies and sits in engineering most of the time. It suffers from noticability, same as Engine engineer.
suggestion: Icon of the talent appearing on the repair interface when talent is in effect, ideally showing % bonus applied

**Pump N Dump**
![image](https://user-images.githubusercontent.com/59519434/204115101-3fa1b8ad-c796-48ad-b04c-6216f4d8f1a3.png)
Good talent, especially on subs with 3 large ballasts. Suffers from noticability.
suggestion: Talent applied icon on ballast repair interface somewhere. Maybe a sound on being applied for a dopamine hit

**Hull Fixer**
![image](https://user-images.githubusercontent.com/59519434/204115144-fcc885ab-b8b2-44e6-afde-544c0968c7be.png)
My favorite mechanic talent. Fixfoam grenades, handheld status monitor and hull repair speed is absolute perfection. I enjoyed all 3 of these things a lot, especially throwing fixfoams behind enemy breachers to seal the leaks behind them and start draining immediately.

**Quick Fixer**
![image](https://user-images.githubusercontent.com/59519434/204115195-e303f931-1c3d-4150-90cf-294ac0c4d3fd.png)
A fun talent, when the speedbuff applies... and it didn't seem to apply often. Seemed like the device needs to be damaged significantly in order to get the buff and because I maintained my machines diligently, I didn't get to enjoy the speed.
suggestion: Have the speedbuff apply more reliably.

**Let it Drain**
![image](https://user-images.githubusercontent.com/59519434/204115218-25f55890-05cd-4610-b83a-77ed297a9ab0.png)
Good talent for the pump recipe. I bothered engineers to craft better batteries for me which is cooperation (especially with the better batteries perk for engies). This talent also makes people consider wiring, which is cool and the limitation to 2 makes sense.

**Robotics**
![image](https://user-images.githubusercontent.com/59519434/204115276-3912cf14-6905-4258-b230-39a2e1876cb0.png)
Biggest disappointment for me. The defense bots kept shooting the crew and even myself, even when there were no enemies in the sub. When the defense bot worked as intended, it still shot allies because it starts shooting before it aims the gun at enemies and it's willing to shoot at the enemy even through people. The bots also do not reliably follow 1 person and it's unclear how to make them follow 1 person or another. They are very powerful against ruin guardians and thalamus cells, however after shooting the paralysis cells it turned on my allies and started shooting at them... I could only manage to control 1 thanks to being able to grab them, cant imagine herding 2 of these annoying fuckers. The bots have potential, but currently they are too dangerous for teammates and owner even in peaceful periods for some reason.

**Miner**
![image](https://user-images.githubusercontent.com/59519434/204115371-7b4cc4e7-f8e3-47fe-a5be-55d7402f8e32.png)
Straightforward talent. I like it. Seen the double output get applied enough times to make it feel worth it, though I wish it made a sound. Cutter speed for ores feels significant.
suggestion: make a sound when the double output bonus applies

**Residual Waste**
![image](https://user-images.githubusercontent.com/59519434/204115388-d9108ef3-963a-4846-bef0-f1858c7a30dc.png)
Feels like a must compared to the Mass Production, which feels almost useless. I wish having a combo of 2 mechanics taking 1 of each of these would be the optimal way. I saw the double output popup from time to time and it was a welcome boost to our treasury.
suggestion: make a sound when the double output bonus applies

**Tool Maintenance**
![image](https://user-images.githubusercontent.com/59519434/204115489-19696dfd-3851-479c-9696-d4c32a0995a4.png)
Straightforward. Feels like an absolute must. 1 quartermaster takes this talent and the rest can take the scrap one.

**Iron Storm**
![image](https://user-images.githubusercontent.com/59519434/204115504-9305e8e8-280b-4933-9e06-14a576ff956d.png)
Disappointing! We already had a tier 3 sub and scrap cannon uses scrap, which intervenes with Scrap Savant that makes you get double resources from scrap. I was struggling keeping up with our ammo consumption and there was no way I would feel comfortable shooting steel or scrap (which often drops ammo resources).

**Ballast Denizen**
![image](https://user-images.githubusercontent.com/59519434/204115576-db63211c-743b-4366-9ba8-7f8d18dec8a0.png)
Good talent. Made me not need a diving mask or suit, so I could comfortably fix most leaks with just my safety harness.

**Modular Repairs**
![image](https://user-images.githubusercontent.com/59519434/204115622-7bbe225b-d28e-4347-947d-ae0f1fd7a409.png)
Crafting repair packs is a big waste since it costs tin, which was always in short supply. I wish the engineer merchants sold at least 10 tin per each outpost they appear in. I haven't crafted a single one.

**I Am That Guy**
![image](https://user-images.githubusercontent.com/59519434/204115786-304944c6-ef00-4934-8535-9ad085375681.png)
Didn't get to use this. Feels ok from description.

**Other talents**
![image](https://user-images.githubusercontent.com/59519434/204115669-afb5321d-17be-48b0-8d24-10c7e69efb37.png)
Can't imagine any build where I'd take this. It's not easy to fix 3 mechanical devices in a round since they aren't many and they deteriorate so little + idk if the bonus would apply if I repair things from for example 90% up compared to repairing fully.

![image](https://user-images.githubusercontent.com/59519434/204115693-110125e1-dd13-452d-a17f-cf6b56ef66e6.png)
Undertaking these missions is relatively rare. I'd have this talent replaced with some other wreck related bonus. For example wrecks have a logbook whether you get a salvage mission or not. Could some sort of nice bonus apply for getting such a logbook onboard (similiar to security detaining enemies onboard the sub with By the Book, but smaller reward maybe)

![image](https://user-images.githubusercontent.com/59519434/204115733-72d15ac5-0aed-4f3c-a405-f14895f332d9.png)
Somebody else took the talent so idk how effective this is. I like the design of it.

![image](https://user-images.githubusercontent.com/59519434/204115748-0ab40beb-3bbd-4c62-a205-0a26abeeb0c0.png)
Might work for some campaigns and not for others, idk. I don't see myself ever using it.

![image](https://user-images.githubusercontent.com/59519434/204115758-3e22dab8-4c9b-461f-b8b9-52629d33d5cb.png)
Seems useless compared to alternative. 10% to get back usually a basic ore feels weak.
Suggestion: Increase fabrication speed as long as you keep crafting the same item.

**Comment by ubertpendragon** [comment]
First off: No tinkering.

With that out of the way

1.  Resource balance

True will open a discussion on resource balance for faction patch


2. Talents

![image](https://user-images.githubusercontent.com/73229309/204116957-c71ee7ec-df4e-40a2-8b2c-3a8c4736b655.png)
Repair pack used to have a xp component on its talent, but as a standalone it's too expensive, the 50% extra repair speed is outdone by a bunch of new global passive repair talents (Morale, Layabout)... It doesn't even eliminate skill checks.
Merging with Safety Harness, increasing output to 4x would be a good solution

![image](https://user-images.githubusercontent.com/73229309/204116905-f728bbd8-39fe-4077-a508-fb62ad066390.png)
Heavy lifting is not a good talent, especially when the alternative is more effective HP and essentially one-shotting mudraptors. Security runs 25% faster when carrying ammo boxes which is useful in a situation of battle/emergency. Carrying crates isn't. I'm really not sure what to do with this.

![image](https://user-images.githubusercontent.com/73229309/204117088-6fe1e13c-e31e-46d3-8317-91bc304511f8.png)
It's not just bad, it's also boring. Even a passive +100% fabricator speed would be better

**Comment by Udrakan** [comment]
I forgot to add:
One thing that I miss on my mechanic is a passive that increases the speed of prying doors open. I'd appreciate that as a general talent, or as a side bonus to some scrap/wreck related talent. Maybe a replacement talent after merging Modular Repairs and Safety First?

**Comment by JeroonBosch** [comment]
Good feedback, thanks for taking the time to type this all out!
